src/server/pachyderm_test.go

```diff
@@ -486,6 +485,116 @@ func TestPipeline(t *testing.T) {
 	require.Equal(t, 2, len(listCommitResponse.CommitInfo))
 }
 
+func TestPipelineTransientFailure(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	t.Parallel()
+
+	c := getPachClient(t)
+
+	// create repos
+	dataRepo := uniqueString("TestPipelineTransientFailure_data")
+	require.NoError(t, c.CreateRepo(dataRepo))
+	// create pipeline
+	pipelineName := uniqueString("pipeline")
+	// this pipeline fails at random
+	require.NoError(t, c.CreatePipeline(
+		pipelineName,
+		"",
+		[]string{"bash"},
+		[]string{
+			"if (( RANDOM % 2 )); then exit 0; else exit 1; fi",
+		},
+		&ppsclient.ParallelismSpec{
+			Strategy: ppsclient.ParallelismSpec_CONSTANT,
+			Constant: 1,
+		},
+		[]*ppsclient.PipelineInput{{
+			Repo:   &pfsclient.Repo{Name: dataRepo},
+			Method: client.MapMethod,
+		}},
+		false,
+	))
+
+	numJobs := 50
+	var commit *pfsclient.Commit
+	var err error
+	for i := 0; i < numJobs; i++ {
+		commit, err = c.StartCommit(dataRepo, "master")
+		require.NoError(t, err)
+		_, err = c.PutFile(dataRepo, commit.ID, "file", strings.NewReader("foo\n"))
+		require.NoError(t, err)
+		require.NoError(t, c.FinishCommit(dataRepo, commit.ID))
+	}
+
+	// Wait for the last job to complete
+	_, err = c.FlushCommit([]*pfsclient.Commit{client.NewCommit(dataRepo, commit.ID)}, nil)
+
+	// this is dumb, but there's a period of time between when the output commit is
+	// finished, and when the job's state is updated.
+	time.Sleep(5 * time.Second)
+
+	jobInfos, err := c.ListJob(pipelineName, nil)
+	require.NoError(t, err)
+	require.Equal(t, numJobs, len(jobInfos))
+
+	var jobStates []interface{}
+	for _, jobInfo := range jobInfos {
+		jobStates = append(jobStates, jobInfo.State)
+	}
+
+	// the jobs should've either succeeded or failed
+	for _, jobState := range jobStates {
+		require.EqualOneOf(t, []interface{}{ppsclient.JobState_JOB_SUCCESS, ppsclient.JobState_JOB_FAILURE}, jobState)
+	}
+	// unless we are super unlucky (1 in 2^(RETRY_LIMIT*numJobs)), at least one job should've succeeded.
+	require.EqualOneOf(t, jobStates, ppsclient.JobState_JOB_SUCCESS)
+	// unless we are super unlucky (1-((1/2)^RETRY_LIMIT))^numJobs, at least one job should've failed.
+	// with RETRY_LIMIT=3 and numJobs=50, this comes down to 0.001
+	require.EqualOneOf(t, jobStates, ppsclient.JobState_JOB_FAILURE)
+}
+
+func TestPipelineThatCrashes(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	t.Parallel()
+
+	c := getPachClient(t)
+
+	// create repos
+	dataRepo := uniqueString("TestPipelineThatCrashes_data")
+	require.NoError(t, c.CreateRepo(dataRepo))
+	// create pipeline
+	pipelineName := uniqueString("pipeline")
+	// this pipeline sleeps.
+	// then we are gonna manually kill a pod and see if the job completes successfully
```

Comment:
Yeah, otherwise we're going to need to setup CI to send you a slack message to go in and do your thing. I think there's examples in the tests of killing pachd pods that you can copy paste.